tests/solidity/test_fees.py

```diff
@@ -0,0 +1,367 @@
+        """
+        # assign all keyword args to self
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+        if target_apr:
+            self.target_apr = target_apr
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        self.bob.budget = self.trade_amount
+        self.bob.wallet.balance = types.Quantity(amount=self.trade_amount, unit=types.TokenType.BASE)
+        self.gary = agent.Agent(wallet_address=2, budget=0)
+        self.block_time = time.BlockTime()
+        self.pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.1,  # 0.1e18, // curveFee
+            redemption_fee_percent=0.1,  # 0.1e18, //flatFee
+            governance_fee_percent=gov_fee,  # 0.5e18, //govFee
+        )
+        super().__init__()
+
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=self.pricing_model,
+            market_state=market_state,
+            block_time=self.block_time,
+            position_duration=time.StretchedTime(
+                days=self.term_length,
+                time_stretch=self.pricing_model.calc_time_stretch(self.target_apr),
+                normalizing_constant=self.term_length,
+            ),
+        )
+        _, wallet_deltas = self.hyperdrive.initialize(
+            wallet_address=self.alice.wallet.address,
+            contribution=self.contribution,
+            target_apr=self.target_apr,
+        )
+        self.alice.wallet.update(wallet_deltas)
+        self.market_state_before_open = self.hyperdrive.market_state.copy()
+
+    def assertAlmostEqual(self, first, second, places=None, msg=None, delta=None):
+        """
+        Assert that first and second are almost equal
+        Using this function to cast to Decimal to avoid type comparison errors without tons of casting
+        """
+        # pylint: disable=too-many-arguments
+        if delta is None:
+            delta = 10**-16
+        super().assertAlmostEqual(Decimal(first), Decimal(second), places=places, msg=msg, delta=delta)
+
+
+def idfn(val):
+    """Custom id function for pytest parametrize"""
+    return f"amount={val:.0f}"
+
+
+def get_gov_fees_accrued(test, market_state=None) -> float:
+    """Get the amount of gov fees that have accrued in the market state"""
+    if market_state:
+        return market_state.gov_fees_accrued * market_state.share_price
+    return test.hyperdrive.market_state.gov_fees_accrued * test.hyperdrive.market_state.share_price
+
+
+def warp(test, time_delta):
+    """Move time forward by time_delta and update the share price to simulate interest"""
+    test.hyperdrive.block_time.set_time(test.hyperdrive.block_time.time + time_delta)
+    test.hyperdrive.market_state.share_price = test.market_state_before_open.share_price * (
```

Comment:
verify that this is indeed exactly the same every time it is done in Solidity

Reply:
all 3 instances of setSharePrice in FeeTest.t.sol are identical:
```
hyperdrive.setSharePrice(
            getPoolInfo().sharePrice.mulDown(
                FixedPointMath.ONE_18 + apr.mulDown(timeDelta)
            )
        );
```

Reply:
Ok, then can you rename this to `advance_time_and_set_share_price` or something like that?

Reply:
I renamed it to `advance_time` in a later commit. that matches solidity. they don't use share price explicitly anymore, so i'd rather leave that part out.